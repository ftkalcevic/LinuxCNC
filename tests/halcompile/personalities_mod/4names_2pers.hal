# personalities are used modulo the number of
# halcompiled personalities

# 4 names, 2 personalities
loadrt logic    names=lg1,lg2,lg3,lg4 personality=0x102,0x204
loadrt lincurve names=lc1,lc2,lc3,lc4 personality=2,3
loadrt bitslice names=bs1,bs2,bs3,bs4 personality=2,3

show pin
show parameter
