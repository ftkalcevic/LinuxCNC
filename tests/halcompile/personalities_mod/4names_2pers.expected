bitslice u32   IN 0 bs1.in
bitslice bit   OUT FALSE bs1.out-00
bitslice bit   OUT FALSE bs1.out-01
bitslice s32   OUT 0 bs1.time
bitslice u32   IN 0 bs2.in
bitslice bit   OUT FALSE bs2.out-00
bitslice bit   OUT FALSE bs2.out-01
bitslice bit   OUT FALSE bs2.out-02
bitslice s32   OUT 0 bs2.time
bitslice u32   IN 0 bs3.in
bitslice bit   OUT FALSE bs3.out-00
bitslice bit   OUT FALSE bs3.out-01
bitslice s32   OUT 0 bs3.time
bitslice u32   IN 0 bs4.in
bitslice bit   OUT FALSE bs4.out-00
bitslice bit   OUT FALSE bs4.out-01
bitslice bit   OUT FALSE bs4.out-02
bitslice s32   OUT 0 bs4.time
lincurve float IN 0 lc1.in
lincurve float OUT 0 lc1.out
lincurve float I/O 0 lc1.out-io
lincurve s32   OUT 0 lc1.time
lincurve float IN 0 lc2.in
lincurve float OUT 0 lc2.out
lincurve float I/O 0 lc2.out-io
lincurve s32   OUT 0 lc2.time
lincurve float IN 0 lc3.in
lincurve float OUT 0 lc3.out
lincurve float I/O 0 lc3.out-io
lincurve s32   OUT 0 lc3.time
lincurve float IN 0 lc4.in
lincurve float OUT 0 lc4.out
lincurve float I/O 0 lc4.out-io
lincurve s32   OUT 0 lc4.time
logic bit   OUT FALSE lg1.and
logic bit   IN FALSE lg1.in-00
logic bit   IN FALSE lg1.in-01
logic s32   OUT 0 lg1.time
logic bit   IN FALSE lg2.in-00
logic bit   IN FALSE lg2.in-01
logic bit   IN FALSE lg2.in-02
logic bit   IN FALSE lg2.in-03
logic bit   OUT FALSE lg2.or
logic s32   OUT 0 lg2.time
logic bit   OUT FALSE lg3.and
logic bit   IN FALSE lg3.in-00
logic bit   IN FALSE lg3.in-01
logic s32   OUT 0 lg3.time
logic bit   IN FALSE lg4.in-00
logic bit   IN FALSE lg4.in-01
logic bit   IN FALSE lg4.in-02
logic bit   IN FALSE lg4.in-03
logic bit   OUT FALSE lg4.or
logic s32   OUT 0 lg4.time

bitslice s32   RW 0 bs1.tmax
bitslice bit   RO FALSE bs1.tmax-increased
bitslice s32   RW 0 bs2.tmax
bitslice bit   RO FALSE bs2.tmax-increased
bitslice s32   RW 0 bs3.tmax
bitslice bit   RO FALSE bs3.tmax-increased
bitslice s32   RW 0 bs4.tmax
bitslice bit   RO FALSE bs4.tmax-increased
lincurve s32   RW 0 lc1.tmax
lincurve bit   RO FALSE lc1.tmax-increased
lincurve float RW 0 lc1.x-val-00
lincurve float RW 0 lc1.x-val-01
lincurve float RW 0 lc1.y-val-00
lincurve float RW 0 lc1.y-val-01
lincurve s32   RW 0 lc2.tmax
lincurve bit   RO FALSE lc2.tmax-increased
lincurve float RW 0 lc2.x-val-00
lincurve float RW 0 lc2.x-val-01
lincurve float RW 0 lc2.x-val-02
lincurve float RW 0 lc2.y-val-00
lincurve float RW 0 lc2.y-val-01
lincurve float RW 0 lc2.y-val-02
lincurve s32   RW 0 lc3.tmax
lincurve bit   RO FALSE lc3.tmax-increased
lincurve float RW 0 lc3.x-val-00
lincurve float RW 0 lc3.x-val-01
lincurve float RW 0 lc3.y-val-00
lincurve float RW 0 lc3.y-val-01
lincurve s32   RW 0 lc4.tmax
lincurve bit   RO FALSE lc4.tmax-increased
lincurve float RW 0 lc4.x-val-00
lincurve float RW 0 lc4.x-val-01
lincurve float RW 0 lc4.x-val-02
lincurve float RW 0 lc4.y-val-00
lincurve float RW 0 lc4.y-val-01
lincurve float RW 0 lc4.y-val-02
logic s32   RW 0 lg1.tmax
logic bit   RO FALSE lg1.tmax-increased
logic s32   RW 0 lg2.tmax
logic bit   RO FALSE lg2.tmax-increased
logic s32   RW 0 lg3.tmax
logic bit   RO FALSE lg3.tmax-increased
logic s32   RW 0 lg4.tmax
logic bit   RO FALSE lg4.tmax-increased

